src/main/java/hudson/plugins/octopusdeploy/OctopusDeployPushRecorder.java

```diff
@@ -0,0 +1,220 @@
+
+        final Iterable<String> patternSplit = Splitter.on("\n")
+                .trimResults()
+                .omitEmptyStrings()
+                .split(packagePathPattern);
+        FilePath ws = workspace;
+        for (final String pattern : patternSplit) {
+            final List<FilePath> matchingFiles = fileService.getMatchingFile(ws, pattern, log);
+            /*
+                Don't add duplicates
+             */
+            for (final FilePath file : matchingFiles) {
+                files.add(file);
+                if (file != null) {
+                    final FilePath existing = CollectionUtils.find(files, new Predicate<FilePath>() {
+                        @Override
+                        public boolean evaluate(FilePath existingFile) {
+                            return existingFile.getBaseName().equals(file.getBaseName());
+                        }
+                    });
+
+                    if (existing == null) {
+                        files.add(file);
+                    }
+                }
+            }
+        }
+
+        try {
+            final List<String> commands = buildCommands(envInjector, files, ws);
+            final Boolean[] masks = getMasks(commands, OctoConstants.Commands.Arguments.MaskedArguments);
+            Result result = launchOcto(workspace, launcher, commands, masks, envVars, listenerAdapter);
+            success = result.equals(Result.SUCCESS);
+        } catch (Exception ex) {
+            log.fatal("Failed to push the packages: " + ex.getMessage());
+            success = false;
+        }
+
+        if (!success) {
+            run.setResult(Result.FAILURE);
+        }
+    }
+
+    private List<String> buildCommands(final EnvironmentVariableValueInjector envInjector, final List<FilePath> files, FilePath workspace) throws IOException, InterruptedException {
+        final List<String> commands = new ArrayList<>();
+
+        OctopusDeployServer server = getOctopusDeployServer(this.serverId);
+        String serverUrl = server.getUrl();
+        String apiKey = server.getApiKey().getPlainText();
+        boolean ignoreSslErrors = server.getIgnoreSslErrors();
+        OverwriteMode overwriteMode = this.overwriteMode;
+        Boolean verboseLogging = this.verboseLogging;
+        String additionalArgs = envInjector.injectEnvironmentVariableValues(this.additionalArgs);
+
+        checkState(StringUtils.isNotBlank(serverUrl), String.format(OctoConstants.Errors.INPUT_CANNOT_BE_BLANK_MESSAGE_FORMAT, "Octopus URL"));
+        checkState(StringUtils.isNotBlank(apiKey), String.format(OctoConstants.Errors.INPUT_CANNOT_BE_BLANK_MESSAGE_FORMAT, "API Key"));
+        checkState(!files.isEmpty(), String.format("The pattern \n%s\n failed to match any files", packagePaths));
+
+        commands.add("push");
+
+        commands.add("--server");
+        commands.add(serverUrl);
+
+        commands.add("--apiKey");
+        commands.add(apiKey);
```

Comment:
This will not mask the fields from the job output - you could also build the arguments by using the hudson [`ArgumentBuilder`](https://javadoc.jenkins.io/hudson/util/ArgumentListBuilder.html)